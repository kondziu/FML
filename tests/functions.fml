// Expected output:
// > left(true, false)    = true
// > right(true, false)   = false
// > neither(true, false) = null
// > scope before: x=1, y=2, z=3
// > scope inside: x=6, y=7, z=3
// > scope after:  x=1, y=2, z=5

function left(x, y) -> x;
function right(x, y) -> y;
function neither(x, y) -> null;

print("left(true, false)    = ~\n", left(true, false));
print("right(true, false)   = ~\n", right(true, false));
print("neither(true, false) = ~\n", neither(true, false));

let x = 1;
let y = 2;
let z = 3;

function my_scope(x, y) ->
begin
   print("scope inside: x=~, y=~, z=~\n", x, y, z);
   y <- 4;
   z <- 5;
end;

print("scope before: x=~, y=~, z=~\n", x, y, z);
my_scope(6, 7);
print("scope after:  x=~, y=~, z=~\n", x, y, z);