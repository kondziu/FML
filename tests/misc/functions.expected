left(true, false)    = true
right(true, false)   = false
neither(true, false) = null
scope before: x=1, y=2, z=3
scope inside: x=6, y=7, z=3
scope after:  x=1, y=2, z=5
